lessons/advanced/behaviours.md

```diff
@@ -0,0 +1,49 @@
+---
+layout: page
+title: Behaviours
+category: advanced
+order: 
+lang: en
+---
+
+We learned about Typespecs the previous lesson, here we'll learn about how to require a module implement those specifications.  In Elixir, this functionality is referred to as Behaviours.
+
+{% include toc.html %}
+
+## Uses
+
+Behaviours give us the ability to define a set of functions a module must implement and ensure that set has been implemented, this makes it easy to create and share a public API.
```

Comment:
What would you think about this?
>Usually, you expect module to share a public API. The native solution for this in Elixir is [behaviour](http://elixir-lang.org/docs/stable/elixir/behaviours.html). Behaviour does only two simple things:
* defines a set of function that must be implemented;
* checks whether that set was actually implemented.